### PR TITLE
Feature/support to crashpad trace

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cmd/pktvisord/main.cpp [hunk 1]
```diff
@@ -68,6 +69,10 @@ static const char USAGE[] =
     Configuration:
       --config FILE               Use specified YAML configuration to configure options, Taps, and Collection Policies
                                   Please see https://pktvisor.dev for more information
+    Crashpad:
+      --cp-token TOKEN            Specifies Crashpad information for tracing issues remote
```
Comment (weyrick) [inline cmd/pktvisord/main.cpp]: "Crashpad token for remote crash reporting"

File: cmd/pktvisord/main.cpp [hunk 2]
```diff
@@ -68,6 +69,10 @@ static const char USAGE[] =
...
+      --cp-url URL                Crashpad server url
+      --cp-path PATH              Crashpad binary handler
```
Comment (weyrick) [inline cmd/pktvisord/main.cpp]: Crashpad handler binary

File: cmd/pktvisord/main.cpp [hunk 3]
```diff
@@ -68,6 +69,10 @@ static const char USAGE[] =
     Configuration:
       --config FILE               Use specified YAML configuration to configure options, Taps, and Collection Policies
                                   Please see https://pktvisor.dev for more information
+    Crashpad:
```
Comment (weyrick) [inline cmd/pktvisord/main.cpp:72]: so we also need a way for users to turn this off, even in cases where they are using a container that hasn't enabled by default. perhaps `--cp-disable` and/or checking an environment variable

File: cmd/pktvisord/CrashpadHandler.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    std::map<std::string, std::string> annotations;
+    std::vector<std::string> arguments;
+    CrashpadClient client;
+    bool rc;
+
+    annotations["format"] = "minidump";
+    annotations["product"] = "pktvisor";
+    annotations["database"] = "pktvisor";
+    annotations["version"] = VISOR_VERSION_NUM;
+    annotations["token"] = token;
```
Comment (weyrick) [inline cmd/pktvisord/CrashpadHandler.h:23]: is other information like OS, kernel or anything else sent automatically?

File: cmd/pktvisord/CMakeLists.txt [hunk 5]
```diff
@@ -2,9 +2,13 @@ add_executable(pktvisord main.cpp)
 
 target_link_libraries(pktvisord
         PRIVATE
+        ${CONAN_LIBS_CRASHPAD}
```
Comment (weyrick) [inline cmd/pktvisord/CMakeLists.txt:5]: so we have to figure out how to make this optional at build time

File: cmd/pktvisord/CrashpadHandler.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    base::FilePath::StringType db_path("crashpad");
```
Comment (weyrick) [inline cmd/pktvisord/CrashpadHandler.h:24]: is this a file on disk? i see "no-rate-limit" below, but we would need to be careful if this writes to disk, we can't let the disk fill

File: cmd/pktvisord/main.cpp [hunk 7]
```diff
@@ -385,6 +415,18 @@ int main(int argc, char *argv[])
...
 
+    // crashpad support
+    if (options.crashpad_info.token.first || options.crashpad_info.url.first || options.crashpad_info.path.first) {
+        if (options.crashpad_info.token.first && options.crashpad_info.url.first && options.crashpad_info.path.first) {
+            if (!crashpad::start_crashpad_handler(options.crashpad_info.token.second, options.crashpad_info.url.second, options.crashpad_info.path.second)) {
+                logger->error("failed to setup crashpad");
+            }
+        }
+        else {
+            logger->error("missing information to setup crashpad");
```
Comment (weyrick) [inline cmd/pktvisord/main.cpp]: in these error cases, should we exit?

**Review by weyrick — CHANGES_REQUESTED** [review]

File: cmd/pktvisord/CrashpadHandler.h (hunk 4 above)
Comment (ghost) [reply cmd/pktvisord/CrashpadHandler.h:23]: Yes it does
```
uname.machine   amd64
uname.release   null
uname.sysname   Linux
uname.version   5.14.0 -1031-oem #34-Ubuntu SMP Fri Mar 25 08:21:05 UTC 2022 x86_64
....
```

File: cmd/pktvisord/main.cpp (hunk 3 above)
Comment (leoparente) [reply cmd/pktvisord/main.cpp:72]: Don't get why should we need to add this disable. if the crashpad configurations are no passed correctly, the crashpad code will not be executed

File: cmd/pktvisord/CMakeLists.txt (hunk 5 above)
Comment (leoparente) [reply cmd/pktvisord/CMakeLists.txt:5]: the code will be not called if config is not passed

File: cmd/pktvisord/CrashpadHandler.h (hunk 6 above)
Comment (leoparente) [reply cmd/pktvisord/CrashpadHandler.h:24]: We can change that on future commit after we get this properly running

File: cmd/pktvisord/main.cpp (hunk 7 above)
Comment (leoparente) [reply cmd/pktvisord/main.cpp]: yes, i will add that

File: cmd/pktvisord/CrashpadHandler.h (hunk 6 above)
Comment (leoparente) [reply cmd/pktvisord/CrashpadHandler.h:24]: the rate limit should be a dump per hour

File: cmd/pktvisord/main.cpp (hunk 3 above)
Comment (weyrick) [reply cmd/pktvisord/main.cpp:72]: If it's only going into the debug container, then that's okay. But if it's going to be enabled by default in the release container, then people need a way to shut it off.

**Review by weyrick — APPROVED** [review]

**Review by weyrick — APPROVED** [review]

**Pull request merged**